Automattic/wp-calypso
# Use VideoPress player on core video blocks

#### Changes proposed in this Pull Request

This PR updates the VideoPress extension for Gutenberg in order to use the VideoPress player on the core video blocks instead of the native HTML5 video player. This enhances the user experience since the VideoPress player includes several built-in features such as toggling HD or displaying a placeholder if the video is being processed. 

#### Tasks

- [x] Display the player on the editor view.
- [x] Display the player on the published view.
- [x] Change the source of the video settings attributes, so they are not lost when the core video blocks are migrated to the customized video block generated by the VideoPress extension.

#### To follow-up in future PRs

- Bring back the block settings (autoplay, loop, etc) and use them in the VideoPress player.

#### Testing instructions



* Select an environment:
  * Gutenlypso.
    * Start Calypso with Calypsoify Iframe disabled: `DISABLE_FEATURES=calypsoify/iframe npm start`.
    * Go to `/block-editor`.
    * Select a site under a premium/business plan.
  * Jetpack.
    * [Create a new JN site using the extensions from this branch](https://jurassic.ninja/create?gutenpack&calypsobranch=update/gutenberg-videopress-player&jetpack-beta&shortlived&wp-debug-log).
    * Connect to WP.com using a premium/business subscription.
    * Go to Settings → Jetpack Constants in the site’s wp-admin and set `JETPACK_BETA_BLOCKS` to true.
    * Go to Jetpack → Settings and activate the VideoPress module (by enabling the "Enable high-speed, ad-free video player" option).
    * Go to Posts → New.
* Insert a video block.
* Upload or select an existing video.
  * Note that in Jetpack sites, VideoPress only converts videos uploaded via the Media Library, so don't use the "Upload" button when testing on Jetpack (bug already reported: #11194).
* Make sure the VideoPress player is previewed on the editor.

<img width="692" alt="screen shot 2019-02-01 at 18 43 12" src="https://user-images.githubusercontent.com/1233880/52139746-40084100-2651-11e9-9cc0-ba41f62f1b5c.png">

* Publish the post.
* Check that the VideoPress player is rendered on the published view.
* Open in the block editor an existing post that contains a video block using the default video player (you can easily do this by creating a post in https://wordpress.com and opening it later in your dev env).
* Confirm that the block is migrated and it uses now the VideoPress player. 
* Switch to the Code view and make sure the video settings (autoplay, loop, muted, ...) are also serialized in the block content.

<img width="595" alt="screen shot 2019-02-07 at 13 08 10" src="https://user-images.githubusercontent.com/1233880/52410934-4b88bb80-2ada-11e9-98f1-6883558f8130.png">
